src/components/screens/EarnScreen.tsx

```
@@ -85,6 +91,24 @@ export function EarnScreen() {
+  const filtered = useMemo(() => {
+    const q = query.trim().toLowerCase();
+    if (!q) return shown;
+    return shown.filter((p) =>
+      p.pair.toLowerCase().includes(q) ||
+      p.dex.toLowerCase().includes(q) ||
+      (p.version ?? '').toLowerCase().includes(q)
+    );
+  }, [shown, query]);
+
+  // Reset paging whenever the result set changes so a new search starts at the top.
+  useEffect(() => { setVisible(PAGE_SIZE); }, [query, pools]);
```

ESLint

[error] Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

/home/jailuser/git/src/components/screens/EarnScreen.tsx:107:21
  105 |
  106 |   // Reset paging whenever the result set changes so a new search starts at the top.
> 107 |   useEffect(() => { setVisible(PAGE_SIZE); }, [query, pools]);
      |                     ^^^^^^^^^^ Avoid calling setState() directly within an effect
  108 |
  109 |   const page = filtered.slice(0, visible);
  110 |   const remaining = filtered.length - page.length;

(react-hooks/set-state-in-effect)